thinc/types.py

```diff
@@ -841,7 +841,7 @@ class Ragged:
     data: Array2d
     lengths: Ints1d
     data_shape: Tuple[int, ...]
-    cumsums: Optional[Ints1d] = None
+    starts_ends: Optional[Ints1d] = None
```

Comment:
Should this be underscored? It shouldn't really be publicly accessed, should it? Now I'm also worrying about bwd-compat if we merge this change, though I don't think anyone would have been accessing this field directly?

Reply:
Yeah, it would probably be better. Interestingly `cumsums` was made public relatively recently in https://github.com/explosion/thinc/commit/4e8cf6f68628f563c509d4cb7b9057fe0b82cc4a.

Reply:
Oh. pinging @adrianeboyd in that case - it looks like there was a good reason to do this.

Reply:
I think Daniël's getting years mixed up, that wasn't that recent?

Reply:
It's possible newer versions of pydantic don't have the same behavior or someone else can figure out how to get it to work, but I don't know any more details than what's mentioned in the PR.

Reply:
Having extra fields is disallowed here (which is probably good?):

https://github.com/explosion/thinc/blob/95b1ec978091431994e6ab8a7b366e177fa4a5f8/thinc/util.py#L427

But it seems like we cannot change this specifically for the `Ragged` class. I am not sure if I understand the issue correctly, but this may be due to:

https://github.com/samuelcolvin/pydantic/issues/986

Reply:
I could add a `cumsums` property to ensure we are not breaking the API.

Reply:
Discussed internally, we can keep this as is. This wasn't really part of the public API and isn't documented as such.